### ISSUE TITLE
0.8.2 > 0.8.3  gives repeating "No space left on device" errors

### ISSUE THREAD

**Issue by jameswarner62** [blakeblackshear/frigate#829]
Upgraded from 0.8.2 to 0.8.3 - same config - I now get errors after a short while and the container needs restarting.

/tmp/cache is showing as full within 30minutes or so (6 cameras) this didn't seem to happen with 0.8.1

ffmpeg.garage.clips_rtmp ERROR : Guessed Channel Layout for Input Stream #0.1 : mono
ffmpeg.garage.clips_rtmp ERROR : Could not write header for output file #0 (incorrect codec parameters ?): No space left on device

**Comment by blakeblackshear** [comment]
Nothing related to this changed from 0.8.2 to 0.8.3. Are you sure it wasn't 0.8.1 > 0.8.3?

That should be talking about writing to `/tmp/cache`.

**Comment by jameswarner62** [comment]
Yes fairly sure - as I posted an issue before (version grabbed from debug page) it was the latest stable from dockerhub before this latest release...
I tried doubling tmp cache to 2gb and limiting clip length down from 300 to 200 seconds. it still filled up overnight, and now needs a restart:
![image](https://user-images.githubusercontent.com/56305354/108820355-ff6f1d80-75b3-11eb-85fc-a4ec7e0b4dfa.png)

The cache has 19,000 clips in....

**Comment by blakeblackshear** [comment]
Can you provide the information requested in the issue template? Logs, config, etc

**Comment by jameswarner62** [comment]
Sure - see below.  going back to 0.8.1 fixes the issue - the cache is maintained at 93-95% at all times but overflows in 0.8.3.

I also notice that one of the cameras is not being segmented just keeps growing, whilst the rest are cycled. When it runs out of space this single file is using  1.9GB of the 2GB cache
```
-rw-r--r-- 1 root root   6512385 Feb 23 13:02 drive-20210223130150.mp4
-rw-r--r-- 1 root root   6512725 Feb 23 13:02 drive-20210223130200.mp4
-rw-r--r-- 1 root root   6505988 Feb 23 13:02 drive-20210223130210.mp4
-rw-r--r-- 1 root root   6517918 Feb 23 13:02 drive-20210223130221.mp4
-rw-r--r-- 1 root root   6510988 Feb 23 13:02 drive-20210223130231.mp4
-rw-r--r-- 1 root root   6506367 Feb 23 13:02 drive-20210223130241.mp4
-rw-r--r-- 1 root root   6508424 Feb 23 13:03 drive-20210223130251.mp4
-rw-r--r-- 1 root root   6538591 Feb 23 13:03 drive-20210223130301.mp4
-rw-r--r-- 1 root root   6481646 Feb 23 13:03 drive-20210223130311.mp4
-rw-r--r-- 1 root root   6544192 Feb 23 13:03 drive-20210223130321.mp4
-rw-r--r-- 1 root root   3407920 Feb 23 13:03 drive-20210223130331.mp4
-rw-r--r-- 1 root root   1158042 Feb 23 13:02 garage-20210223130151.mp4
-rw-r--r-- 1 root root   1193584 Feb 23 13:02 garage-20210223130201.mp4
-rw-r--r-- 1 root root   1183353 Feb 23 13:02 garage-20210223130211.mp4
-rw-r--r-- 1 root root   1163495 Feb 23 13:02 garage-20210223130221.mp4
-rw-r--r-- 1 root root   1170444 Feb 23 13:02 garage-20210223130231.mp4
-rw-r--r-- 1 root root   1190576 Feb 23 13:02 garage-20210223130241.mp4
-rw-r--r-- 1 root root   1187049 Feb 23 13:03 garage-20210223130251.mp4
-rw-r--r-- 1 root root   1178346 Feb 23 13:03 garage-20210223130301.mp4
-rw-r--r-- 1 root root   1183739 Feb 23 13:03 garage-20210223130311.mp4
-rw-r--r-- 1 root root   1184161 Feb 23 13:03 garage-20210223130321.mp4
-rw-r--r-- 1 root root    524336 Feb 23 13:03 garage-20210223130331.mp4
-rw-r--r-- 1 root root 525074480 Feb 23 13:03 garden-20210223123525.mp4
-rw-r--r-- 1 root root   3226649 Feb 23 13:02 house-20210223130150.mp4
-rw-r--r-- 1 root root   3169988 Feb 23 13:02 house-20210223130202.mp4
-rw-r--r-- 1 root root   1643889 Feb 23 13:02 house-20210223130214.mp4
-rw-r--r-- 1 root root   3298836 Feb 23 13:02 house-20210223130220.mp4
-rw-r--r-- 1 root root   3220150 Feb 23 13:02 house-20210223130232.mp4
-rw-r--r-- 1 root root   1478521 Feb 23 13:02 house-20210223130244.mp4
-rw-r--r-- 1 root root   3227283 Feb 23 13:03 house-20210223130250.mp4
-rw-r--r-- 1 root root   3506224 Feb 23 13:03 house-20210223130302.mp4
-rw-r--r-- 1 root root   1659765 Feb 23 13:03 house-20210223130314.mp4
-rw-r--r-- 1 root root   3251475 Feb 23 13:03 house-20210223130320.mp4
-rw-r--r-- 1 root root   1310768 Feb 23 13:03 house-20210223130332.mp4
-rw-r--r-- 1 root root   5183868 Feb 23 13:02 road-20210223130150.mp4
-rw-r--r-- 1 root root   6249638 Feb 23 13:02 road-20210223130200.mp4
-rw-r--r-- 1 root root   6523600 Feb 23 13:02 road-20210223130210.mp4
-rw-r--r-- 1 root root   6158506 Feb 23 13:02 road-20210223130220.mp4
-rw-r--r-- 1 root root   6568934 Feb 23 13:02 road-20210223130230.mp4
-rw-r--r-- 1 root root   6564191 Feb 23 13:02 road-20210223130240.mp4
-rw-r--r-- 1 root root   6212020 Feb 23 13:03 road-20210223130250.mp4
-rw-r--r-- 1 root root   4841305 Feb 23 13:03 road-20210223130300.mp4
-rw-r--r-- 1 root root   5739468 Feb 23 13:03 road-20210223130310.mp4
-rw-r--r-- 1 root root   5135973 Feb 23 13:03 road-20210223130320.mp4
-rw-r--r-- 1 root root   4194352 Feb 23 13:03 road-20210223130330.mp4
-rw-r--r-- 1 root root   1164629 Feb 23 13:01 workshop-20210223130145.mp4
-rw-r--r-- 1 root root    582734 Feb 23 13:02 workshop-20210223130157.mp4
-rw-r--r-- 1 root root   1166670 Feb 23 13:02 workshop-20210223130203.mp4
-rw-r--r-- 1 root root   1170303 Feb 23 13:02 workshop-20210223130215.mp4
-rw-r--r-- 1 root root    580816 Feb 23 13:02 workshop-20210223130227.mp4
-rw-r--r-- 1 root root   1164308 Feb 23 13:02 workshop-20210223130233.mp4
-rw-r--r-- 1 root root   1165641 Feb 23 13:02 workshop-20210223130245.mp4
-rw-r--r-- 1 root root    588159 Feb 23 13:03 workshop-20210223130257.mp4
-rw-r--r-- 1 root root   1080072 Feb 23 13:03 workshop-20210223130303.mp4
-rw-r--r-- 1 root root   1080756 Feb 23 13:03 workshop-20210223130315.mp4
-rw-r--r-- 1 root root    541214 Feb 23 13:03 workshop-20210223130327.mp4
-rw-r--r-- 1 root root    262192 Feb 23 13:03 workshop-20210223130333.mp4
```


**Version of frigate**
0.8.3

**Config file**
Include your full config file wrapped in triple back ticks.
```yaml
detectors:
  coral:
    type: edgetpu
    device: usb

mqtt:
  host: xxxxx
  topic_prefix: frigate
  user: xxxx
  password: "xxxx"

logger:
  default: debug

clips:
  max_seconds: 200
  tmpfs_cache_size: 2G
  retain:
    default: 10

ffmpeg:
  global_args: -hide_banner -loglevel warning
  hwaccel_args: -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p
  input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclock_as_timestamps 1

  output_args:
    detect: -f rawvideo -pix_fmt yuv420p
    record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    clips: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    rtmp: -c copy -f flv
objects:
  track:
    - person
    - car
    - dog
    - cat
  filters:
    person:
      min_area: 4000
      max_area: 100000
      threshold: 0.65

cameras:
  drive:
    ffmpeg:
      inputs:
        - path: rtsp://xxxx.245:554/stream1
          roles:
            - detect
        - path: rtsp://xxxx.245:554/stream0
          roles:
            - rtmp
            - clips


    height: 576
    width: 704
    fps: 5
    clips:
      enabled: True
      pre_capture: 5
      post_capture: 5
      retain:
        default: 14
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True

    objects:
      track:
        - person
        - car
      filters:
        person:
          min_area: 500
          max_area: 100000
          threshold: 0.75
        car:
          min_area: 500
          max_area: 100000
          threshold: 0.75
          mask:
            - 704,201,704,576,0,576,0,330,370,311,424,197

  road:
    ffmpeg:
      inputs:
        - path: rtsp://xxxx.241:554/stream2
          roles:
            - detect
        - path: rtsp://xxxx.241:554/stream1
          roles:
            - rtmp
            - clips
    clips:
      enabled: True
      pre_capture: 5
      post_capture: 5
      retain:
        default: 14
    height: 480
    width: 640
    fps: 10
    motion:
      delta_alpha: 0.3
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
    objects:
      track:
        - person
        - car
      filters:
        person:
          min_area: 1000
          max_area: 100000
          threshold: 0.75
        car:
          min_area: 3000
          max_area: 100000
          threshold: 0.7

  house:
    ffmpeg:
      inputs:
        - path: rtsp://xxxx.240:554/stream2
          roles:
            - detect
        - path: rtsp://xxxx.240:554/stream1
          roles:
            - rtmp
            - clips
    height: 480
    width: 640
    fps: 5
    motion:
      mask:
        - 0,158,0,0,640,0,640,258,640,350,640,480,363,480,512,342,521,326,524,121
    clips:
      enabled: True
      pre_capture: 5
      post_capture: 5
      retain:
        default: 14
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
    objects:
      track:
        - person
        - car
        - dog
        - cat
      filters:
        person:
          min_area: 500
          max_area: 100000
          threshold: 0.7
        car:
          min_area: 500
          max_area: 100000
          threshold: 0.75
          mask:
            - 0,204,0,271,219,266,221,191

  workshop:
    ffmpeg:
      inputs:
        - path: rtsp://xxxx.238:554/stream2
          roles:
            - detect
        - path: rtsp://xxxx.238:554/stream1
          roles:
            - rtmp
            - clips
    height: 480
    width: 640
    fps: 5
    clips:
      enabled: True
      pre_capture: 5
      post_capture: 5
      retain:
        default: 14
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
    objects:
      track:
        - person
      filters:
        person:
          min_area: 500
          max_area: 100000
          threshold: 0.65

  garage:
    ffmpeg:
      inputs:
        - path: rtsp://admin:xxxx@xxxx.247:80/ch0_1.264
          roles:
            - detect
        - path: rtsp://admin:xxxx@xxxx.247:80/ch0_0.264
          roles:
            - rtmp
            - clips

    height: 360
    width: 640
    fps: 5
    clips:
      enabled: True
      pre_capture: 5
      post_capture: 5
      retain:
        default: 14
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
    objects:
      track:
        - person
      filters:
        person:
          min_area: 500
          max_area: 100000
          threshold: 0.75


  garden:
    ffmpeg:
      inputs:
        - path: rtsp://xxxx.237:554/stream2
          roles:
            - detect
        - path: rtsp://xxxx.237:554/stream1
          roles:
            - rtmp
            - clips
    height: 480
    width: 640
    fps: 5
    clips:
      enabled: True
      pre_capture: 5
      post_capture: 5
      retain:
        default: 14
    motion:
      contour_area: 50
      delta_alpha: 0.6
      threshold: 15
      mask:
        - 0,480,640,480,640,364,640,265,640,0,588,0,574,170,463,148,76,217,101,91,263,54,285,0,0,0
        - 0,0,640,0,640,385,558,390,566,72,243,56,0,99
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
    objects:
      track:
        - person
        - dog
        - cat
      filters:
        person:
          min_area: 500
          max_area: 100000
          threshold: 0.65
          mask:
            - 91,410,297,440,498,452,605,444,640,430,640,251,640,203,510,207

```

**Frigate container logs**
```
Include relevant log output here
```

**Frigate stats**
```json
{
"detection_fps": 1.1,
"detectors": {
"coral": {
"detection_start": 0,
"inference_speed": 8.08,
"pid": 37
}
},
"drive": {
"camera_fps": 5,
"capture_pid": 46,
"detection_fps": 0,
"pid": 40,
"process_fps": 5,
"skipped_fps": 0
},
"garage": {
"camera_fps": 5.1,
"capture_pid": 56,
"detection_fps": 0,
"pid": 44,
"process_fps": 5.1,
"skipped_fps": 0
},
"garden": {
"camera_fps": 5.1,
"capture_pid": 61,
"detection_fps": 1.1,
"pid": 45,
"process_fps": 5,
"skipped_fps": 0
},
"house": {
"camera_fps": 5.1,
"capture_pid": 51,
"detection_fps": 0,
"pid": 42,
"process_fps": 5.1,
"skipped_fps": 0
},
"road": {
"camera_fps": 10.1,
"capture_pid": 47,
"detection_fps": 0,
"pid": 41,
"process_fps": 10.1,
"skipped_fps": 0
},
"service": {
"storage": {
"/dev/shm": {
"free": 1068.6,
"mount_type": "tmpfs",
"total": 1073.7,
"used": 5.2
},
"/media/frigate/clips": {
"free": 1722292.8,
"mount_type": "fuse.shfs",
"total": 3000348.7,
"used": 1278055.9
},
"/media/frigate/recordings": {
"free": 1722292.8,
"mount_type": "fuse.shfs",
"total": 3000348.7,
"used": 1278055.9
},
"/tmp/cache": {
"free": 0,
"mount_type": "tmpfs",
"total": 2147.5,
"used": 2147.5
}
},
"uptime": 7256,
"version": "0.8.3-d771726"
},
"workshop": {
"camera_fps": 5.1,
"capture_pid": 55,
"detection_fps": 0,
"pid": 43,
"process_fps": 5.1,
"skipped_fps": 0
}
}
```

**FFprobe from your camera**

Run the following command and paste output below
```
Input #0, rtsp, from 'rtsp://xxxx241:554/stream1':
  Metadata:
    title           : stream server
    comment         : stream1
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: h264, yuvj420p(pc, progressive), 2304x1536, 40 tbr, 90k tbn, 180k tbc
    Stream #0:1: Audio: pcm_mulaw, 8000 Hz, 1 channels, s16, 64 kb/s
```


**Computer Hardware**
 - OS: Unraid
 - Install method: unraid docker
 - Virtualization: 
 - Coral Version: USB
 - Network Setup: Wired

**Camera Info:**
 - Manufacturer: Mixed
 - Model: Mixed
 - Resolution: 2-4MP
 - FPS: [e.g. 5] 5fps detect

**Comment by jameswarner62** [comment]
Anything else I can do to help troubleshoot? If the segmentation issue is not easily resolved, or is an ffmpeg issue - can you look at adding a cache clear that limits individual file size? ie: if any cache file grows beyond 500MB then force delete/restart that capture process?

**Comment by blakeblackshear** [comment]
We need to figure out why ffmpeg can't split the video into segments for those cameras. If you post the output of /api/config, I can help create an ffmpeg command you can run outside of frigate to see what is happening.

**Comment by jameswarner62** [comment]
This issue appears resolves in 0.9b1
